pkg/app/api/grpcapi/piped_api.go

```diff
@@ -1206,13 +1210,33 @@ func (a *PipedAPI) InChainDeploymentPlannable(ctx context.Context, req *pipedser
 		return nil, status.Error(codes.InvalidArgument, "invalid deployment with chain block index provided")
 	}
 
-	prevBlockSuccessfullyCompleted, err := dc.IsSuccessfullyCompletedBlock(req.DeploymentChainBlockIndex - 1)
-	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, "unable to process the previous block of this deployment in chain")
+	previousBlock := dc.Blocks[req.DeploymentChainBlockIndex-1]
```

Comment:
Excuse me, I'm missing something. Can you brief me on why the given `DeploymentChainBlockIndex` needs to be decreased? The index starts with zero, doesn't it?

Reply:
we need the status of the previous block to determine should we start planning deployments of this block or not, that's why I named the variable as `previousBlock` and the code handler for `blockIndex = 0` is above this part [L1201-L1207](https://github.com/pipe-cd/pipe/pull/2866/files/c9b7200220c2e9d5582c61b7d803ef7fa3f68283#diff-dc8d28c3b055fcad0f022406b744ce756e3be8c630b6c40aef12e026c6a58dcfR1201-R1207)

Reply:
Oh yeah, I totally forgot!!